students/sewelol/main.go

```diff
@@ -0,0 +1,159 @@
+package main
+
+import (
+	"bufio"
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"math/rand"
+	"os"
+	"strconv"
+	"strings"
+	"time"
+)
+
+// Problem structure
+type Problem struct {
+	q string
+	a int
+}
+
+// PROBLEMBUFCOUNT sets number of problems in buffer
+const PROBLEMBUFCOUNT = 100
+
+// DEFAULTTIMELIMIT sets the default time limit for the quiz
+const DEFAULTTIMELIMIT = 30 //seconds
+
+// Problem counter
+var count int
+
+// Score counter
+var score int
+var faults int
+
+// readProblems takes reads problems from file, line by line
+// problems are written to problems channel
+func readProblems(problems chan Problem, filename string, shuffle bool) {
+	// problems buffer
+	buf := make([]Problem, PROBLEMBUFCOUNT)
+
+	// Open problems file
+	fd, err := os.Open(filename)
+	if err != nil {
+		log.Fatal(err)
+	}
+	defer fd.Close()
+
+	// Scan file
+	scanner := bufio.NewScanner(fd)
+	for scanner.Scan() {
+		// Read comma separated line
+		line := strings.Split(scanner.Text(), ",")
+		// first value is the question string
+		q := line[0]
+		// Convert string answer to integer
+		ans, err := strconv.Atoi(line[1])
+		if err != nil {
+			log.Fatal(err)
+		}
+
+		// store problem in buffer
+		buf[count] = Problem{q, ans}
+		count++
+	}
+
+	// shuffle problems
+	if shuffle {
+		rand.Seed(time.Now().Unix())
+		for i := range buf[:count] {
+			j := rand.Intn(i + 1)
+			buf[i], buf[j] = buf[j], buf[i]
+		}
+	}
+
+	// send problems over channel
+	for _, p := range buf[:count] {
+		problems <- p
+	}
+
+}
+
+// solveProblem consumes problems from chan
+// will check timer before giving points
+func solveProblem(problems chan Problem) {
```

Comment:
If you wanted, both `solveProblem` and `readProblem` could use directional channel arguments to help clarify how they use the channel. See https://gobyexample.com/channel-directions for an example.